Update testing-before-launch.mdx (#5955)
dfinity/portal: docs/building-apps/governing-apps/tokenomics/sns-checklist.mdx

Provide the build and deploy instructions with the source code. Ideally the instructions are a part of the code repository README file, and if that’s not the case, a link to the instructions should be available in the README file. In order to be able to create a reproducible build, the build environment needs to be reproducible. The [documentation](/docs/building-apps/best-practices/reproducible-builds) provides more information how reproducible builds can be created using Docker.

## 2.4. Test the SNS launch 

> It is recommended to conduct a local test of the SNS launch process. By doing so, you can simulate the complete SNS initial token swap process also from the user’s perspective via the NNS frontend dapp. [View detailed instructions on how to set up SNS locally for testing are available](https://github.com/dfinity/sns-testing).

Reviewer: ```suggestion
It is recommended to conduct a local test of the SNS launch process. By doing so, you can simulate the complete SNS initial token swap process also from the user’s perspective via the NNS frontend dapp. [View detailed instructions on how to set up SNS locally for testing are available](https://github.com/dfinity/ic/blob/master/rs/sns/testing/README.md).
```

## 2.5. Test dapp operations

> In addition to testing the launch, before requesting an SNS launch in production, developers are strongly encouraged to ensure that all operations for maintaining and evolving the dapp can be done via SNS proposals.

Reviewer: ```suggestion
In addition to testing the launch, before requesting an SNS launch in mainnet, developers are strongly encouraged to ensure that all operations for maintaining and evolving the dapp can be done via SNS proposals.
```

Make sure to test upgrading canisters through SNS proposals, test updating asset canister content through SNS proposals, and other typical upgrade and maintenance operations. Also establish a [cycles management strategy](../managing/cycles-usage.mdx), so canisters never run out of cycles. 
[View information on how to test an locally](https://github.com/dfinity/sns-testing).

## 2.6. Integrate an SNS frontend into the dapp

> Developers can choose to integrate a frontend for the SNS functionality in the dapp. A good example of a useful integration is SNS proposal voting. This allows neurons to vote on proposals directly in the dapp frontend. Integrations should be tested thoroughly, for example by [local testing](https://github.com/dfinity/sns-testing).

Reviewer: ```suggestion
Developers can choose to integrate a frontend for the SNS functionality in the dapp. A good example of a useful integration is SNS proposal voting. This allows neuron holders to vote on proposals directly in the dapp frontend. Integrations should be tested thoroughly, for example by [local testing](https://github.com/dfinity/sns-testing).
```
